### ISSUE TITLE
packAssociativeArrays does not work on recursive arrays

### ISSUE THREAD

**Issue by sebastian-lenz** [craftcms/cms#6320]
### Description

It looks like `ProjectConfig::packAssociativeArray` copies the first found recursive value into all subsequent child nodes with recursive arrays.

### Steps to reproduce

The following test code illustrates the issue. It passes an associative array with nested associated arrays into the pack / unpack functions. We would expect the end result to be equal to the original input, however it is not. Note how the second child item transforms into the first value.

```php
$value = [
    'rootA' => [
        'label' => 'childA'
    ],
    'rootB' => [
        'label' => 'childB'
    ]
];
print json_encode($value);

$value = \craft\helpers\ProjectConfig::packAssociativeArray($value);
print json_encode($value);

$value = \craft\helpers\ProjectConfig::unpackAssociativeArray($value);
print json_encode($value);
```

Output:
```json
{
  "rootA": {"label": "childA"},
  "rootB": {"label": "childB"}
}
{
  "__assoc__":[
    ["rootA", {"__assoc__": [["label", "childA"]]}],
    ["rootB", {"__assoc__": [["label", "childA"]]}]
  ]
}
{
  "rootA": {"label": "childA"},
  "rootB": {"label": "childA"}
}
```

### Additional info

- Craft version: Craft Pro 3.4.26
- PHP version: 7.4.6
- Database driver & version: MySQL 5.7.29

**Comment by sebastian-lenz** [comment]
The issue seems to stem from the usage of the value reference here:
https://github.com/craftcms/cms/blob/develop/src/helpers/ProjectConfig.php#L223

I've crated a pull request that replaces the loop with a call to `array_map` which seems to resolve the issue in my test case.

**Comment by brandonkelly** [comment]
This was already reported by #6317. We just released Craft 3.4.27 with a fix.

**Comment by darylknight** [comment]
Got it - thanks for pull requesting!
